Mathlib/Topology/Algebra/Valued/LocallyCompact.lean

```diff
@@ -148,4 +151,133 @@ lemma totallyBounded_iff_finite_residueField [IsDiscreteValuationRing 𝒪[K]] :
 
 end FiniteResidueField
 
+section CompactDVR
+
+open Valued
+
+variable (K) in
+lemma exists_norm_coe_lt : ∃ x : 𝒪[K], 0 < ‖(x : K)‖ ∧ ‖(x : K)‖ < 1 := by
+  obtain ⟨x, hx, hx'⟩ := NormedField.exists_norm_lt_one K
+  refine ⟨⟨x, hx'.le⟩, ?_⟩
+  simpa [hx', Subtype.ext_iff] using hx
+
+variable (K) in
+lemma exists_norm_lt : ∃ x : 𝒪[K], 0 < ‖x‖ ∧ ‖x‖ < 1 := by
+  exact exists_norm_coe_lt K
+
+variable (K) in
+lemma exists_nnnorm_lt : ∃ x : 𝒪[K], 0 < ‖x‖₊ ∧ ‖x‖₊ < 1 := by
+  exact exists_norm_coe_lt K
+
+lemma isDiscreteValuationRing_of_compactSpace [h : CompactSpace 𝒪[K]] :
```

Comment:
This proof is over 70 lines long. It's a tour de force, it compiles quickly, the code is great. But I *personally* would love to see a sketch of the mathematics going on underneath, perhaps in comments interspersed in the code. I feel like this would be an easy job for you, given that you wrote the code: could I persuade you to do this? Mathlib usually doesn't document long proofs and I think this is bad.

Reply:
How does it look now?